### PR TITLE
Manage banner in editor and update banner in song select

### PR DESCRIPTION
**Pull request by Kizuna-dev**
Use a 3 options dialog instead so that the user can choose if they want to change the background or the banner.
When adding banner it will get added to all diff of the mapset.
Reload banner so it get instantly updated to the new one when coming back in the song select screen.

New button image added here : https://github.com/Quaver/Quaver.Resources/pull/11

https://github.com/Quaver/Quaver/issues/3192

### CODE REVIEW

File: Quaver.Shared/Assets/UserInterface.cs [hunk 1]
```diff
@@ -114,6 +114,8 @@ public static class UserInterface
         public static Texture2D ClosePanel => TextureManager.Load(@"Quaver.Resources/Textures/UI/SongSelect/close-panel.png");
         public static Texture2D CreateButton => TextureManager.Load(@"Quaver.Resources/Textures/UI/create-button.png");
         public static Texture2D SureButton => TextureManager.Load(@"Quaver.Resources/Textures/UI/sure-button.png");
+        public static Texture2D BackgroundButton => TextureManager.Load(@"Quaver.Resources/Textures/UI/background-button.png");
```
[the remark this answers is not in the document]
Comment (Kizuna-dev) [reply Quaver.Shared/Assets/UserInterface.cs:117]: Done, made it yellow-ish

File: Quaver.Shared/Screens/Edit/Dialogs/EditorChangeBackgroundBannerDialog.cs [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+using Wobble.Logging;
+
+namespace Quaver.Shared.Screens.Edit.Dialogs
+{
+    public class EditorChangeBackgroundBannerDialog : BackgroundBannerDialog
+    {
+        public EditorChangeBackgroundBannerDialog(EditScreen screen, string file) : base("CHANGE BACKGROUND / BANNER",
+            $"Would you like to change the background or the banner?")
+        {
+            BackgroundAction += () =>
```
[the remark this answers is not in the document]
Comment (Kizuna-dev) [reply Quaver.Shared/Screens/Edit/Dialogs/EditorChangeBackgroundBannerDialog.cs]: Done i think ? not sure if that was what you asked, but looks better to me

File: Quaver.Shared/Screens/Edit/Dialogs/EditorChangeBackgroundBannerDialog.cs [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+                    }
+                    catch (Exception)
+                    {
+                        // ignored
+                    }
+
+                    screen.WorkingMap.BannerFile = name;
+                    screen.Map.BannerPath = name;
+                    screen.Save(true, true);
+
```
[the remark this answers is not in the document]
Comment (Kizuna-dev) [reply Quaver.Shared/Screens/Edit/Dialogs/EditorChangeBackgroundBannerDialog.cs]: Not sure if that's "invalidating" but i made it so it remove the previous one that way it update with the new one

**Comment by Swan** [comment]
Please have a look at https://github.com/Quaver/Quaver/pull/3251 for more information about what I meant. It is a full rewrite of this feature.

**Pull request closed without merging**